src/main/python/mmlspark/cyber/ml/access_anomalies/collaborative_filtering.py

```diff
@@ -0,0 +1,721 @@
+    creates the AccessAnomalyModel which is a pyspark.ml.Transformer
+    """
+    tenantCol = Param(
+        Params._dummy(),
+        "tenantCol",
+        "The name of the tenant column. "
+        "This is a unique identifier used to partition the dataframe into independent "
+        "groups where the values in each such group are completely isolated from one another. "
+        "Note: if this column is irrelevant for your data, "
+        "then just create a tenant column and give it a single value for all rows."
+    )
+
+    userCol = Param(
+        Params._dummy(),
+        "userCol",
+        "The name of the user column. "
+        "This is a the name of the user column in the dataframe."
+    )
+
+    resCol = Param(
+        Params._dummy(),
+        "resCol",
+        "The name of the resource column. "
+        "This is a the name of the resource column in the dataframe."
+    )
+
+    likelihoodCol = Param(
+        Params._dummy(),
+        "likelihoodCol",
+        "The name of the column with the likelihood estimate for user, res access "
+        "(usually based on access counts per time unit). "
+    )
+
+    outputCol = Param(
+        Params._dummy(),
+        "outputCol",
+        "The name of the output column representing the calculated anomaly score. "
+        "Values will be between (-inf, +inf) with an estimated mean of 0.0 and standard deviation of 1.0. "
+    )
+
+    rankParam = Param(
+        Params._dummy(),
+        "rankParam",
+        "rankParam is the number of latent factors in the model (defaults to 10)."
+    )
+
+    maxIter = Param(
+        Params._dummy(),
+        "maxIter",
+        "maxIter is the maximum number of iterations to run (defaults to 25)."
+    )
+
+    regParam = Param(
+        Params._dummy(),
+        "regParam",
+        "regParam specifies the regularization parameter in ALS (defaults to 0.1)."
+    )
+
+    numBlocks = Param(
+        Params._dummy(),
+        "numBlocks",
+        "numBlocks is the number of blocks the users and items will be partitioned into "
+        "in order to parallelize computation "
+        "(defaults to |tenants| if separate_tenants is False else 10)."
+    )
+
+    separateTenants = Param(
+        Params._dummy(),
+        "separateTenants",
+        "separateTenants applies the algorithm per tenant in isolation. "
+        "Setting to True may reduce runtime significantly, if number of tenant is large, "
+        "but will increase accuracy. (defaults to False)."
+    )
+
+    lowValue = Param(
+        Params._dummy(),
+        "lowValue",
+        "lowValue is used to scale the values of likelihood_col to be in the range [lowValue, highValue] "
+        "(defaults to 5.0)."
+    )
+
+    highValue = Param(
+        Params._dummy(),
+        "highValue",
+        "highValue is used to scale the values of likelihood_col to be in the range [lowValue, highValue] "
+        "(defaults to 10.0)."
+    )
+
+    algoCfParams = Param(
```

Comment:
break this into its regular parameters and add here

Reply:
Will be fixed by removing CfAlgoParams

Reply:
done